gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -365,69 +282,131 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
                     kspTask.setSource(kotlinCompilation.defaultSourceSet.kotlin)
                 }
             }
+            kspTask.exclude { kspOutputDir.isParentOf(it.file) }
```

Comment:
this line seems to be new, can you elaborate it?

Reply:
Generated sources do not participate in processing. Previously, this was done in overridden `sources` in KSP tasks. BTW, `javaSources` is still needed because the `patternFilterable` isn't implemented properly in KGP.